gke-deploy/README.md

```diff
@@ -0,0 +1,205 @@
+for more on how to use the Git Resource.
+
+See
+[here](https://github.com/tektoncd/pipeline/blob/master/docs/auth.md#ssh-authentication-git)
+to learn how to access private Git repositories using Tekton.
+
+
+# `build-push-gke-deploy` `Pipeline`
+
+This Pipeline builds, pushes, and deploys your application to a Google Kubernetes Engine cluster using [`gke-deploy`](https://github.com/GoogleCloudPlatform/cloud-builders/tree/master/gke-deploy).
+
+## Install the Pipeline
+
+```
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/gke-deploy/build-push-gke-deploy.yaml
+```
+
+## Inputs
+
+### Resources
+
+* **source-repo**: The Git repository that contains your application's Dockerfile and Kubernetes configs.
+
+### Parameters
+
+* **pathToContext**: The path to the build context relative to your source repo's root. This is used by Kaniko.
+
+  _default_: `.`
+
+* **pathToDockerFile**: The path to the dockerfile to build, relative to the context.
+
+  _default_: `Dockerfile`
+
+* **pathToYamlConfigs**: The path to the Kubernetes configs to deploy, relative to your source repo's root.
+
+* **imageUrl**: URL of image repository.
+
+* **imageTag**: Tag to apply to the built image.
+
+* **clusterName**: Name of target GKE cluster to deploy to.
+
+* **clusterLocation**: Zone/region of target GKE cluster to deploy to.
+
+* **clusterProject**: Project of target GKE cluster to deploy to.
+
+  _default_: `""`
+
+  If no value is provided, the target cluster is assumed to be in the same project as the cluster running this Pipeline.
+
+## Usage
+
+### Authorizing the Pipeline
+
+See [Authorizing `gcloud`
+commands](../gcloud/README.md#authorizing-gcloud-commands) for authorizing the
+`build-push-gke-deploy` Pipeline using a Google IAM Service Account (the instructions will
+describe authorizing `gcloud`, but the steps are the same).
+
+In order for the `build-push-gke-deploy` Pipeline to push an image to your project, add the [`roles/storage.admin`](https://cloud.google.com/iam/docs/understanding-roles#storage-roles) role to the Google service account that the Pipeline runs as. This will allow the Pipeline to push an image to your project, as defined [here](https://cloud.google.com/container-registry/docs/access-control).
+
+```bash
+gcloud iam service-accounts [IMAGE_REGISTRY_PROJECT] add-iam-policy-binding \
+  --role roles/storage.admin \
+  --member "serviceAccount:[SERVICE_ACCOUNT_EMAIL]"
+```
+
+In order for the `build-push-gke-deploy` Pipeline to deploy to a GKE cluster in your project,
+the service account that the Task runs as must have the
+[`roles/container.developer`](https://cloud.google.com/iam/docs/understanding-roles#kubernetes-engine-roles) role.
+
+```bash
+gcloud iam service-accounts [DEPLOY_CLUSTER_PROJECT] add-iam-policy-binding \
+  --role roles/container.developer \
+  --member "serviceAccount:[SERVICE_ACCOUNT_EMAIL]"
+```
+
+### Running the Pipeline
```

Comment:
Can we also demo a setup in which one pipeline docs `build-push`, and another triggers on push of the GCR image and does `deploy`? That would scale more readily to a multi-env setup, and is also closer to what we recommend to users.

Reply:
Yes, we can do that in a future PR. I have one ready but it won't work until the [script](https://github.com/tektoncd/pipeline/blob/master/docs/tasks.md#step-script) feature is released.

Reply:
I'll work on this in a future PR.